src/Pages/Dashboard/UploadData.jsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import Papa from "papaparse";
+import { apply, compose, lift, splitAt, zipObj } from "ramda";
+
+const timestamp = Date.now();
+
+class UploadData extends React.Component {
+  sendData = data => {
+    this.props.firebase
+      .database()
+      .ref(this.props.sensorId + "/" + timestamp)
+      .set(data);
+  };
+
+  handleFile = event => {
+    const file = event.target.files[0];
+    const keys = this.props.keys;
+    const onDataUploaded = this.props.onDataUploaded;
+    const onError = this.props.onError;
+    const sendData = this.sendData;
+
+    Papa.parse(file, {
+      skipEmptyLines: true,
+      error: function(err, file, inputElem, reason) {
+        onError({ err, file, inputElem, reason });
+      },
+      complete: function(results) {
+        const data = results.data;
+
+        // remove display headers
+        data.shift();
+
+        // add api headers
+        data.unshift(keys);
+
+        // convert all values in the 2D array to float
+        for (var i in data.slice(1)) {
+          for (var j in data.slice(1)[i]) {
+            data.slice(1)[i][j] = parseFloat(data.slice(1)[i][j]);
+            console.log(data.slice(1)[i][j]);
+          }
+        }
+
+        // convert arrays to objects
+        const formatedResult = compose(apply(lift(zipObj)), splitAt(1))(data);
```

Comment:
If you use `step` to collect and format all your values (into proper objects), then you can use `complete` as a simple handler to update your values in firebase.